bouture.js

```diff
@@ -59,30 +60,42 @@ function completeElement (tag, args) {
         Object.keys(arg)
           .forEach(attributeName => {
             const attributeValue = arg[attributeName]
-            switch (typeof attributeValue) {
-              case 'boolean':
-                if (attributeValue) {
-                  element.setAttribute(attributeName, '')
+            // check for event binding
+            const eventName = attributeName.match(/^on/) && attributeName.replace(/^once|on/, '')
```

Comment:
This works but doesn't feel super future-proof. What if a non-event attribute starts with `on-`? (is it already the case?)

The problem being solved here is deciding whether `attributeName` is an event or an attribute. I feel a dedicated function would make this clearer.
I don't know how [react](https://github.com/facebook/react) does it, but it'd be interesting to look at how they cut the mustard (and why they ended up with this conclusion)

Reply:
I tried to find how react solved it, and I failed.